ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -386,17 +402,38 @@ Tensor convert_python_tensor_to_tt_tensor(
         TT_THROW("The argument must be of type torch.Tensor or numpy.ndarray!");
     }
 
+    // TODO: Remove check of num_elements from python against volume of ttnn::SimpleShape
+    TT_FATAL(
+        num_elements == shape.volume(),
+        "Number of elements from python tensor {} must match volume of shape {}!",
+        num_elements,
+        shape.volume());
+
+    Layout layout;
```

Comment:
please initialize with some value else it is undefined and can easily strike

Reply:
maybe
```
Layout layout = optional_layout.value_or(Layout::ROW_MAJOR)
// force_to_be_tile_if_type_requires
...
```

Reply:
will fix